src/Features/Core/Portable/Completion/CompletionServiceWithProviders.cs

```diff
@@ -220,6 +227,7 @@ private CompletionProvider GetProviderByName(string providerName)
                     if (ShouldTriggerCompletion(text, caretPosition, trigger, roles, options))
                     {
                         triggeredProviders = providers.Where(p => p.ShouldTriggerCompletion(text, caretPosition, trigger, options)).ToImmutableArrayOrEmpty();
+                        Debug.Assert(ValidatePossibleTriggerCharacterSet(triggeredProviders, document, text, caretPosition));
```

Comment:
hrmm... this runs in the deletion case.  should we only validate if this is an insertion?